PostcodeSearchBuilderBlobTrigger/search.py

```diff
@@ -0,0 +1,172 @@
+import logging
+import requests
+import json
+import os
+
+from . import exceptions
+from . import models
+
+
+def build_index(url, api_key, api_version, index_name):
+
+    try:
+        index = Index(url, api_key, api_version, index_name)
+
+        index.delete_if_already_exists()
+        index.create()
+    except Exception:
+        raise
+
+
+def load_index(url, api_key, api_version, index_name, rows):
+
+    try:
+        load = Load(url, api_key, api_version, index_name, rows)
+
+        load.postcode_documents()
+    except Exception:
+        raise
+
+
+class Index():
+    """Creates a new index"""
+    def __init__(self, url, api_key, api_version, index_name):
+
+        self.query_string = '?api-version=' + api_version
+        self.index_name = index_name
+        self.url = url
+
+        self.headers = {
+            'Content-Type': 'application/json',
+            'api-key': api_key,
+            'odata': 'verbose'
+        }
+
+    def delete_if_already_exists(self):
+
+        try:
+            delete_url = self.url + "/indexes/" + self.index_name + \
+                self.query_string
+
+            response = requests.delete(delete_url, headers=self.headers)
+
+        except requests.exceptions.RequestException as e:
+            logging.exception('unexpected error deleting index', exc_info=True)
+            raise exceptions.StopEtlPipelineErrorException(e)
+
+        if response.status_code == 404:
+            logging.warning(
+                f'postcode search index not found, unable to delete index\
+                           prior to recreation\n index: {self.index_name}')
+
+        elif response.status_code != 204:
+            # 204 is the expected response, because normally there
+            # will always be a postcode search index that already exists
+            logging.error(
+                f'unexpected response when deleting existing search index,\
+                            search_response: {response.json()}\nindex-name:\
+                            {self.index_name}\nstatus: {response.status_code}')
+
+            raise exceptions.StopEtlPipelineErrorException
+
+    def create(self):
+        self.get_index()
+
+        try:
+            create_url = self.url + "/indexes" + self.query_string
+            response = requests.post(create_url,
+                                     headers=self.headers,
+                                     json=self.postcode_schema)
+
+        except requests.exceptions.RequestException as e:
+            logging.exception('unexpected error creating index', exc_info=True)
+            raise exceptions.StopEtlPipelineErrorException(e)
+
+        if response.status_code != 201:
+            logging.error(f'failed to create search index\n\
+                            index-name: {self.index_name}\n\
+                            status: {response.status_code}')
+
+            raise exceptions.StopEtlPipelineErrorException
+
+    def get_index(self):
```

Comment:
As this doesn't return anything, would a better name be build_.... e.g., build_postcode_schema?